src/model/page-config-resolver.js

```diff
@@ -157,7 +172,7 @@ class JsonLdParser {
         continue;
       }
       element[ALREADY_SEEN] = true;
-      if (element.textContent.indexOf('NewsArticle') == -1) {
+      if (!RE_ALLOWED_TYPES.test(element.textContent)) {
```

Comment:
is there a reason this cannot use checkType() as well?

Reply:
using `checkType` will require parsing json which is expensive (publisher can include entire article text in ld+json markup so it can be quite long).  This is an optimization to see if our allowed types are even present before we go in ad to a real parse.  It only matters if swg.js is included on pages that are not our type so it may be unneeded. WDYT?